dune/http.go

```diff
@@ -24,20 +109,61 @@ func decodeBody(resp *http.Response, dest interface{}) error {
 
 func httpRequest(apiKey string, req *http.Request) (*http.Response, error) {
 	req.Header.Add("X-DUNE-API-KEY", apiKey)
-	resp, err := http.DefaultClient.Do(req)
-	if err != nil {
-		return nil, fmt.Errorf("failed to send request: %w", err)
-	}
+	p := defaultRetryPolicy
+	attempt := 1
+	for {
+		resp, err := http.DefaultClient.Do(req)
+		if err != nil {
+			if attempt >= p.MaxAttempts {
+				return nil, fmt.Errorf("failed to send request: %w", err)
+			}
+			time.Sleep(nextBackoff(attempt, p))
+			attempt++
+			continue
+		}
+
+		if resp.StatusCode == 200 {
+			return resp, nil
+		}
 
-	if resp.StatusCode != 200 {
 		defer resp.Body.Close()
-		var errorResponse ErrorResponse
-		err := json.NewDecoder(resp.Body).Decode(&errorResponse)
-		if err != nil {
-			return nil, fmt.Errorf("failed to read error response body: %w", err)
+		snippetBytes, _ := io.ReadAll(io.LimitReader(resp.Body, 1024))
+		var er ErrorResponse
+		_ = json.Unmarshal(snippetBytes, &er)
```

Comment:
This unmarshal could fail, since the body is capped at 1024 characters, so you'll need to handle the error here in that case